src/v/cluster_link/source_topic_syncer.cc

```diff
@@ -344,15 +348,26 @@ source_topic_syncer::run_impl(ss::abort_source&) {
       std::views::keys(candidates_for_update),
       std::back_inserter(topics_to_describe));
 
+    if (as.abort_requested()) {
+        co_return state_transition{
+          .desired_state = model::task_state::stopped,
+          .reason = "Task aborted during run_impl"};
+    }
+
     kafka::describe_configs_response response;
     try {
         response = co_await describe_topics(
           cluster,
           controller_id.value(),
           describe_configs_version,
           topics_to_describe,
-          _config.get_topic_properties_to_mirror());
+          _config.get_topic_properties_to_mirror(),
+          as);
```

Comment:
I think it would be a lot cleaner like this:
```suggestion
    try {
        as.check();
        auto request = make_describe_topics_request(
          topics_to_describe, _config.get_topic_properties_to_mirror());
        response = co_await cluster.dispatch_to(std::move(req));
          controller_id, std::move(request), describe_configs_version);
```